### PR TITLE
Update the release-1.16 release branch from upstream for KFNBC changes

### PR DESCRIPTION
**Pull request by LaVLaS**
Updates the `release-1.16` release branch with the upstream odh-dashboard commits for the RHODS 1.16 release.  This also includes the @laubai changes included in #224

Jira Issues:

- https://issues.redhat.com/browse/RHODS-2740
- https://issues.redhat.com/browse/RHODS-3624
- https://issues.redhat.com/browse/RHODS-4032
- https://issues.redhat.com/browse/RHODS-4938
- https://issues.redhat.com/browse/RHODS-4950

Rhods livebuild for these changes with KFNBC and migration script: [quay.io/llasmith/rhods-operator-live-catalog:1.16.0-dashboard-kfnbc](https://quay.io/llasmith/rhods-operator-live-catalog:1.16.0-dashboard-kfnbc)

### CODE REVIEW

**Comment by bdattoma** [comment]
@LaVLaS I don't think `[Add group config panel & support for multi-groups (](https://github.com/red-hat-data-services/odh-dashboard/pull/230/commits/c0713354122f57436fca3dfbddeb0035b6dddfd0)https://github.com/opendatahub-io/odh-dashboard/pull/270[)](https://github.com/red-hat-data-services/odh-dashboard/pull/230/commits/c0713354122f57436fca3dfbddeb0035b6dddfd0)` should be merged, since there's not enough time for QE to validate it @mwaykole @lucferbux

**Review by andrewballantyne — APPROVED** [review]

**Comment by andrewballantyne** [comment]
cc @cfchase 

@bdattoma I'm tagging Chris here. He made a strong argument for needing this for 1.16. I'll let him comment. Apologizes for the late manifest of what was going into 1.16. I hope we can work to get a more forward looking list for the next release.

File: manifests/overlays/odhdashboardconfig/odh-dashboard-config.yaml [hunk 1]
```diff
@@ -40,3 +40,6 @@ spec:
       requests:
         cpu: "4"
         memory: 4Gi
+  groupsConfig:
+    adminGroups: 'odh-admins'
```
Comment (mwaykole) [inline manifests/overlays/odhdashboardconfig/odh-dashboard-config.yaml:44]: i am not sure if this should be rhods-admin , is it correct ?

Comment (bdattoma) [reply manifests/overlays/odhdashboardconfig/odh-dashboard-config.yaml:44]: I think it should be dedicated-admins as for previous rhods versions

**Comment by bdattoma** [comment]
> anifest of what was going into 1.1

don't worry @andrewballantyne :) anyway, validating this new feature takes time and effort (we also need to automate tests) especially because it's an important one. That's why we should get the RHODS live build at the beginning of the third week at least. Moreover, given the importance, if any bugs is found, it may likely be a blocker.

**Comment by andrewballantyne** [comment]
Understood, thanks @bdattoma -- I'll do my best to continue improving on getting the Dashboard out with more time for QE to test. I'm applying improvements each release, just not as fast as I would like 🙂 If you can think of anything that you'd like as we approach a release upstream for you guys downstream to know about, my (virtual) door is always open, let's talk.

**Comment by bdattoma** [comment]
> Understood, thanks @bdattoma -- I'll do my best to continue improving on getting the Dashboard out with more time for QE to test. I'm applying improvements each release, just not as fast as I would like  If you can think of anything that you'd like as we approach a release upstream for you guys downstream to know about, my (virtual) door is always open, let's talk.

thank you Andrew :)

File: manifests/overlays/odhdashboardconfig/odh-dashboard-config.yaml (hunk 1 above)
Comment (LaVLaS) [reply manifests/overlays/odhdashboardconfig/odh-dashboard-config.yaml:44]: Just so you are aware. The manifests in this repo are not the final manifest that is used for a RHODS deployment.  The official odh-dashboard deployment files will still be located in https://github.com/red-hat-data-services/odh-manifests/tree/master/odh-dashboard AND the RHODS runtime configurations for the ODHDashboardConfig will be deployed via https://github.com/red-hat-data-services/odh-deployer.

**Review by bdattoma — APPROVED** [review]
/qe-approve

**Pull request merged**